### ISSUE TITLE
Add WSL on SCT install documentation for developer

### ISSUE THREAD

**Issue by RignonNoel** [spinalcordtoolbox/spinalcordtoolbox#3642]
### Description

As a developer I tried to follow the contributing section of the repository to follow the official process and begin my journey.

But as I am a Windows user that work with WSL I got stuck in the process and found out all the correct informations only after some research in the [Sphinx documentation](https://spinalcordtoolbox.com/user_section/installation/windows.html)

Maybe could we move the `Installing SCT for development` inside the Sphinx documentation in order to mutualise effort on the different OS flavor and just add some extra steps if needed for people that want to contribute after install ?

### Steps to Reproduce

0. Go to the `README.md` of the repository (most readed page to begin as a developer)

1. Click on the link that redirect to the [contributing section](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Contributing)

2. Follow the steps until the section [Installing SCT for development](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Contributing:-Installing-SCT-for-development)

3. You will be asked to follow steps that can't work with Windows Subsystem for Linux (WSL)

**Expected behavior:** Contributor should have a documentation based on different environment of work (Mac, WSL, etc..) or a text that describe the limitation

**Actual behavior:** Documentation only work for native Linux or people that have advanced knowledge about WSL and Windows intern communication (file sharing, GUI connection, etc..)

**Comment by RignonNoel** [comment]
@joshuacwnewton If you have some directives I can work on it since it's fresh on my head.

**Comment by joshuacwnewton** [comment]
Thank you so much for opening this issue. I think we've only had Linux/Mac developers on our team so far, so your perspective is very valuable here. :)

> Maybe could we move the `Installing SCT for development` inside the Sphinx documentation in order to mutualise effort on the different OS flavor and just add some extra steps if needed for people that want to contribute after install ?

One thing I should mention is that our WSL are already quite complicated. Many of our Windows users are not very tech-savvy, and we've run into issues in the past where our typical Windows users were confused by the instructions! :sweat_smile: 

There is actually another issue open to try and make the WSL instructions simpler and easier to use... (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3610).

So, we would just need to make sure that we don't make things too confusing for our non-developer users. :thinking:

**Comment by joshuacwnewton** [comment]
Sorry for my last comment! I misread what you said. I'm just rewriting my comment now.

**Comment by joshuacwnewton** [comment]
> But as I am a Windows user that work with WSL I got stuck in the process and found out all the correct informations only after some research in the [Sphinx documentation](https://spinalcordtoolbox.com/user_section/installation/windows.html)
> 
> Maybe could we move the `Installing SCT for development` inside the Sphinx documentation in order to mutualise effort on the different OS flavor and just add some extra steps if needed for people that want to contribute after install ?
> ###

My apologies, but I am a bit confused about the link you shared here. It says "Sphinx documentation", but the link points to the Windows documentation. 

Which part of the instructions exactly did you get stuck on?

**Comment by RignonNoel** [comment]
I said `Sphinx documentation` in order to differentiate it from the `Github Wiki`. I was stuck on the Github Wiki because this one is not working for WSL users, and it's only after some troubles that I began research in the issues and found out the WSL install instructions on `spinalcordtoolbox.com`.

I created this ticket for two main reasons:

 1. New contributor from the `README.md` is not working for WSL users (as describe in the `steps to reproduce` of this issue)
 
 2. I think that having developer installation procedure on the Github Wiki lead to to copy/paste from the official installation procedure for normal user since most of the steps are the same, it's why I proposed to discuss the idea of creating kind of a (1) `base install procedure` and (2) `next step for people that want to contribute` to avoid replicate base setup for developer AND not developer

**Comment by joshuacwnewton** [comment]
Given that we now have a native Windows installation method, giving dev instructions for Windows is a little easier, since the installation procedure is more straightforward than it was for WSL.

So, I've rewritten the [Installing SCT for development](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Contributing:-Installing-SCT-for-development) page a bit in order to give explicit dev instructions for Windows users, which I think mostly resolves this issue.

> 2\. I think that having developer installation procedure on the Github Wiki lead to to copy/paste from the official installation procedure for normal user since most of the steps are the same, it's why I proposed to discuss the idea of creating kind of a (1) `base install procedure` and (2) `next step for people that want to contribute` to avoid replicate base setup for developer AND not developer

Thinking about this, I've come up with a few issues with adding a "next step" at the end of the user-focused instructions:

- Developer installations involve a `fork` and `git clone`, while user installations require downloading the source archive, or in the Windows case, downloading an install script. Trying to account for both methods would make the instructions a little more confusing.
- Our developer instructions include _many_ additional steps after the main installation, and I'm worried that appending these to then end might cause the user-focused pages to become even more cluttered/overwhelming than they are now.

So, for the sake of simplicity for our users, I think I'd prefer keeping the dev-focused instructions in the dev Wiki, even if it means duplicating some information from the user-focused pages.

**Comment by mguaypaq** [comment]
+1 from me for linking each page to the other, rather than melding them together. An additional point against a "next steps" approach is that we encourage users in general to say "yes" to the Sentry telemetry question at the start of the installer, whereas we tell developers to say "no". I _think_ it can be disabled pretty easily, but it highlights that we do want to distinguish "pristine" installations from modified installations; a developer install isn't just an "upgraded" version of a regular install.
